src/lwt/win/lwt_unix_impl.ml

```diff
@@ -340,8 +343,9 @@ external start_read :
 external start_write :
   file_descr -> buffer -> int -> int -> int -> unit = "win_write"
 
-let read ch s pos len =
+let read ch b pos len =
 if !d then Format.eprintf "Start reading@.";
+  let s = (Bytes.to_string b) in
```

Comment:
I don't see how this can work. You are creating a copy of b, reading into that copy and then discarding the copy. You have to use directly `read`'s argument.

Looking at the wider picture, it seems you have to change the string argument of `blit_buffer_to_string` to bytes (blitting to a string does not make sense with safe strings). And, possibly, rename that function to `blit_buffer_to_bytes` to avoid confusion.

Reply:
I'll attempt this, but I will probably need some help.
I know essentially zero about OCaml syntax, so seeing the bigger picture is basically by guess and inference for me.

Reply:
It does seem to be working as currently written. Synchronizations are completing correctly.

Reply:
It might work, but it's not correct or future-proof.

Reply:
> It might work, but it's not correct or future-proof.

Not arguing that point, but I don't have the OCaml knowledge to design a correct replacement.